src/python/antares_xpansion/config_loader.py

```diff
@@ -508,44 +509,47 @@ def _set_last_simulation_name(self):
+        self._last_study = self.last_modified_study(self.antares_output())
 
-        if self.step() == "resume":
-            self._simulation_name = last_dir_path
+        if self.step() in ["resume", "sensitivity"] :
+            
+            self._xpansion_simulation_name = self._last_study
+            if self.is_zip(self._last_study):
+                self._xpansion_simulation_name = self._last_study.parent / self._last_study.stem
+                with zipfile.ZipFile(self._last_study, 'r') as output_zip:
+                    output_zip.extractall(self._xpansion_simulation_name)
         else:
-            # Get list of all dirs only in the given directory
-            list_of_zip_filter = Path(self.antares_output()).glob("*.zip")
-
-            # Sort list of files based on last modification time in ascending order
-            list_of_zip = sorted(
-                list_of_zip_filter,
-                key=lambda x: os.path.getmtime(
-                    os.path.join(self.antares_output(), x)),
-            )
-            self._last_zip = list_of_zip[-1]
-            self._simulation_name = self._last_zip.parent / \
-                (self._last_zip.stem+"-Xpansion")
-
-    def is_zip(self, file):
-        filename, ext = os.path.splitext(file)
-        return ext == ".zip"
-
-    def get_last_modified_dir(self, root_dir):
+            self._xpansion_simulation_name = self._last_study.parent / \
+                (self._last_study.stem+"-Xpansion")
+
+    def is_zip(self, study):
+        _, ext = os.path.splitext(study)
+        return ext == ".zip" 
+    
+    def update_last_study_with_sensitivity_results(self):
+        if self.is_zip(self._last_study):
+            os.remove(self._last_study)
+            shutil.make_archive(self._last_study.parent / self._last_study.stem, 'zip', self._xpansion_simulation_name)  
+            if(os.path.exists(self._xpansion_simulation_name)):
+                shutil.rmtree(self._xpansion_simulation_name)
+
+    def is_antares_study(self, study):
+        _, ext = os.path.splitext(study)
+        return ext == ".zip" or os.path.isdir(study)
+
+    def last_modified_study(self, root_dir)-> Path: 
         list_dir = os.listdir(root_dir)
-        list_of_zip = filter(
-            lambda x: self.is_zip(x), list_dir
+        list_of_studies = filter(
+            lambda x: self.is_antares_study(os.path.join(root_dir, x)), list_dir
         )
         # Sort list of files based on last modification time in ascending order
-        zip_sorted = sorted(
-            list_of_zip,
+        sort_studies = sorted(
+            list_of_studies,
             key=lambda x: os.path.getmtime(
                 os.path.join(root_dir, x)),
         )
-
-        last_zip = Path(root_dir) / zip_sorted[-1]
-        filename, ext = os.path.splitext(last_zip)
-        output_dir = os.path.basename(filename)
-        return output_dir
+        last_study = Path(root_dir) / sort_studies[-1]
```

Comment:
We often have unclear error messages here, like `index out of range` if the list `sort_studies` is empty. Is it possible to add a more clear message like `No Antares output has been found` or something equivalent ?